date_range/tests/test_date_range_generator.py

```diff
@@ -49,19 +50,10 @@ def test_generate(self):
         self.assertEqual(range4.type_id, self.type)
 
     def test_generator_multicompany_1(self):
-        generator = self.generator.new({
-            'date_start': '1943-01-01',
-            'name_prefix': '1943-',
-            'type_id': self.typeB.id,
-            'duration_count': 3,
-            'unit_of_time': MONTHLY,
-            'count': 4,
-            'company_id': self.company.id,
-        })
-        generator._cache.update(generator._convert_to_cache(
-            {'company_id': self.company_2.id}, update=True))
-        generator._onchange_company_id()
-        self.assertFalse(generator.type_id)
+        generator_form = Form(self.generator)
+        generator_form.type_id = self.typeB
+        generator_form.company_id = self.company_2
+        self.assertFalse(generator_form.type_id)
```

Comment:
@mreficent Here we don't test the UI, we test the business logic. We therefore don't need the form. All this works wig plain ORM.